### PR TITLE
Bug 1401783 - Use simple matching for extensions.

### PR DESCRIPTION
**Pull request by rnewman**

### CODE REVIEW

File: Extensions/SendTo/Info.plist [hunk 1]
```diff
@@ -33,7 +33,12 @@
 		<key>NSExtensionAttributes</key>
 		<dict>
 			<key>NSExtensionActivationRule</key>
-			<string>SUBQUERY ( extensionItems, $extensionItem, SUBQUERY ( $extensionItem.attachments, $attachment, ANY $attachment.registeredTypeIdentifiers UTI-CONFORMS-TO &quot;public.url&quot;).@count &gt; 0 ).@count == 1</string>
+			<dict>
+				<key>NSExtensionActivationDictionaryVersion</key>
+				<integer>2</integer>
```
Comment (rnewman) [inline Extensions/SendTo/Info.plist]: This basically means "match this against any attachment". The v1 behavior was "pass if this matches every attachment".

**Comment by st3fan** [comment]
It would be really nice if this worked well. Before we land this we just need to make sure our extensions work correctly from Firefox, Safari, Chrome, Twitter & a few other that share URL data. Specially make sure the behaviour is the same - that we get both title and url.

**Comment by garvankeeley** [comment]
FYI, the 'compiled' query in the failing case looks like this:
```
`SUBQUERY ( extensionItems, , SUBQUERY ( , , ANY  UTI-CONFORMS-TO "public.url").@count == 1 ).@count == 1`
```
Those empty parameters look like the problem.

**Comment by garvankeeley** [comment]
@st3fan can we land on master and ask QA to run through this extensively?

**Pull request merged**